core/src/main/scala/akka/kafka/scaladsl/Processor.scala

```diff
@@ -0,0 +1,44 @@
+package akka.kafka.scaladsl
+
+import akka.kafka.ProducerMessage.Envelope
+import akka.kafka.scaladsl.Consumer.DrainingControl
+import akka.kafka.{CommitterSettings, ConsumerSettings, ProducerSettings, Subscription}
+import akka.stream.Materializer
+import akka.stream.scaladsl.Keep
+import akka.{Done, NotUsed}
+import org.apache.kafka.clients.consumer.ConsumerRecord
+
+import scala.concurrent.Future
+
+object Processor {
+
+  def atLeastOnce[K, V](
+      consumerSettings: ConsumerSettings[K, V],
+      subscription: Subscription,
+      committerSettings: CommitterSettings
+  )(process: ConsumerRecord[K, V] => Future[Done])(implicit materializer: Materializer): DrainingControl[Done] = {
+    Consumer
+      .sourceWithOffsetContext(consumerSettings, subscription)
+      .mapAsync(1)(process)
+      .toMat(Committer.sinkWithOffsetContext(committerSettings))(Keep.both)
+      .mapMaterializedValue(DrainingControl.apply)
+      .run()
```

Comment:
And if we return a `Source` instead of running it?

I find it less flexible to immediately run it. In Akka Projections, we may want to wrap the  Source on a `RestartSource` for instance.

Reply:
oh, I understand it now. It's called `Processor` because it's effectively running it.

Reply:
I'm not happy with the naming. I leave that to when we know what we need.

I made it a sink and `run` as you said you just needed the control in https://github.com/akka/akka-projection/pull/15/files#r401689690